### PR TITLE
PLANET-7446: Force no lightbox

### PR DESCRIPTION
**Pull request by mardelnet**
Ref: https://jira.greenpeace.org/browse/PLANET-7446

<hr>

**DESCRIPTION:**
This PR allows editors to remove the lightbox that is added by default to images in the core image block, the gallery block, the media and text block, and inline images of the paragraph block. In order to do that, the `force-no-lightbox` class has to be added to the corresponding block.

Edit-Page-“Test-Lightbox”-‹-Greenpeace-—-WordPress

<hr>

**TESTING:**
https://www-dev.greenpeace.org/test-tavros/test-lightbox/

<hr>

**RELATED PR:**
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/1259

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [tavros](https://www-dev.greenpeace.org/test-tavros) | [admin](https://www-dev.greenpeace.org/test-tavros/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-tavros/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-tavros) | [composer-local.json](https://github.com/greenpeace/planet4-test-tavros/blob/main/composer-local.json)

:watch: 2024.10.30 14:34:51

**Review by mleray — COMMENTED** [review]
I see that the cursor behaviour is not consistent in all these use cases, sometimes it's `pointer` when there's no lightbox (Gallery for example) or it's not `pointer` when needed (inline images for example)

**Comment by mleray** [comment]
But also this PR would not be needed if we end up going with your [native implementation](https://github.com/greenpeace/planet4-master-theme/pull/2428), right?

**Comment by mardelnet** [comment]
> I see that the cursor behaviour is not consistent in all these use cases, sometimes it's `pointer` when there's no lightbox (Gallery for example) or it's not `pointer` when needed (inline images for example)

Fixed!

**Comment by mardelnet** [comment]
> But also this PR would not be needed if we end up going with your [native implementation](https://github.com/greenpeace/planet4-master-theme/pull/2428), right?

The design team decided not to proceed with the native lightbox since it doesn't implement navigation between slides.

**Review by mleray — COMMENTED** [review]
Images no longer show full size in the lightbox, they are rendered there at the same size at they are in the page itself... I've added different images to your [page](https://www-dev.greenpeace.org/test-tavros/test-lightbox/) for testing

**Review by mleray — COMMENTED** [review]
This PR doesn't address the ticket's main issue, which is that we don't respect the "Expand on click" setting. We use the `force-no-caption` class instead. 
If possible we should respect that setting, if not maybe there is a way to remove it using `theme.json`?

**Comment by mardelnet** [comment]
> Images no longer show full size in the lightbox, they are rendered there at the same size at they are in the page itself... I've added different images to your [page](https://www-dev.greenpeace.org/test-tavros/test-lightbox/) for testing

It seems that this problem is unrelated to this PR, as you can see in [this instance](https://www-dev.greenpeace.org/test-rhea/another-test-page/). Apparently, when the image of an Image block is expanded it doesn't use the original size, but the one selected by the editor. Maybe this can be fixed in a follow-up PR.

**Comment by mardelnet** [comment]
> This PR doesn't address the ticket's main issue, which is that we don't respect the "Expand on click" setting. We use the `force-no-caption` class instead. If possible we should respect that setting, if not maybe there is a way to remove it using `theme.json`?

@comzeradd 
This PR addresses many different changes since the solution seems more complex than we initially thought from the ticket description. In a previous conversation with @mleray we agreed on the following items, but before moving on with them and merging the changes, we would like to know your opinion.

In summary, this PR:
- Receives all the logic of the custom lightbox that was removed from the P4 plugin ([check this PR](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/1259)).
- Allows editors to remove the lightbox that is added by default to images in the core image block, the gallery block, the media and text block, and inline images of the paragraph block by adding the `force-no-lightbox` class to the corresponding block (check this PR description for additional details).
- Removes the native "Expand on click" button WordPress adds only to the Image block, in order to use only one solution (adding the `force-no-lightbox`) and avoid confusion among the web editors.

In a follow-up PR, we want to implement a checkbox (or similar) to the blocks sidebar to let editors decide whether to expand images on click, instead of manually adding the `force-no-lightbox` class.
Any thoughts?

**Comment by mleray** [comment]
> It seems that this problem is unrelated to this PR, as you can see in [this instance](https://www-dev.greenpeace.org/test-rhea/another-test-page/). Apparently, when the image of an Image block is expanded it doesn't use the original size, but the one selected by the editor. Maybe this can be fixed in a follow-up PR.

@mardelnet can you please open a bug ticket for this? 🙏 Pretty sure it used to work 🤔

**Comment by comzeradd** [comment]
> In a follow-up PR, we want to implement a checkbox (or similar) to the blocks sidebar to let editors decide whether to expand images on click, instead of manually adding the `force-no-lightbox` class.

Yes, as discussed, let's remove the native toggle and just add the `force-no-lightbox` css class for now.

File: theme.json [hunk 1]
```diff
@@ -179,7 +179,13 @@
...
           ]
         }
-      }
+      },
+      "core/image": {
+				"lightbox": {
+          "enabled": false,
+					"allowEditing": false
+				}
+			}
```
Comment (comzeradd) [inline theme.json:188]: Looks like you are adding tabs here for indendation 😁 Check out [editorconfig](https://greenpeace-gpi.slack.com/archives/C0160MX64AG/p1729754027541789)

**Review by mleray — APPROVED** [review]
Looks good to me, except for Nikos' comment about the indentation in `theme.json` 👍 Tested on the instance and all works as expected 👏

**Comment by mardelnet** [comment]
> > It seems that this problem is unrelated to this PR, as you can see in [this instance](https://www-dev.greenpeace.org/test-rhea/another-test-page/). Apparently, when the image of an Image block is expanded it doesn't use the original size, but the one selected by the editor. Maybe this can be fixed in a follow-up PR.
> 
> @mardelnet can you please open a bug ticket for this? 🙏 Pretty sure it used to work 🤔

Done @mleray 
https://jira.greenpeace.org/browse/PLANET-7649

File: theme.json (hunk 1 above)
Comment (mardelnet) [reply theme.json]: Fixed!

**Pull request merged**